src/packet/ip.rs

```diff
@@ -481,3 +482,167 @@ impl PrimitiveValues for IpNextHeaderProtocol {
+            &IpNextHeaderProtocols::PrivEncryption => "PrivEncryption", // 99
+            &IpNextHeaderProtocols::Gmtp => "Gmtp", // 100
+            &IpNextHeaderProtocols::Ifmp => "Ifmp", // 101
+            &IpNextHeaderProtocols::Pnni => "Pnni", // 102
+            &IpNextHeaderProtocols::Pim => "Pim", // 103
+            &IpNextHeaderProtocols::Aris => "Aris", // 104
+            &IpNextHeaderProtocols::Scps => "Scps", // 105
+            &IpNextHeaderProtocols::Qnx => "Qnx", // 106
+            &IpNextHeaderProtocols::AN => "AN", // 107
+            &IpNextHeaderProtocols::IpComp => "IpComp", // 108
+            &IpNextHeaderProtocols::Snp => "Snp", // 109
+            &IpNextHeaderProtocols::CompaqPeer => "CompaqPeer", // 110
+            &IpNextHeaderProtocols::IpxInIp => "IpxInIp", // 111
+            &IpNextHeaderProtocols::Vrrp => "Vrrp", // 112
+            &IpNextHeaderProtocols::Pgm => "Pgm", // 113
+            &IpNextHeaderProtocols::ZeroHop => "ZeroHop", // 114
+            &IpNextHeaderProtocols::L2tp => "L2tp", // 115
+            &IpNextHeaderProtocols::Ddx => "Ddx", // 116
+            &IpNextHeaderProtocols::Iatp => "Iatp", // 117
+            &IpNextHeaderProtocols::Stp => "Stp", // 118
+            &IpNextHeaderProtocols::Srp => "Srp", // 119
+            &IpNextHeaderProtocols::Uti => "Uti", // 120
+            &IpNextHeaderProtocols::Smp => "Smp", // 121
+            &IpNextHeaderProtocols::Sm => "Sm", // 122
+            &IpNextHeaderProtocols::Ptp => "Ptp", // 123
+            &IpNextHeaderProtocols::IsisOverIpv4 => "IsisOverIpv4", // 124
+            &IpNextHeaderProtocols::Fire => "Fire", // 125
+            &IpNextHeaderProtocols::Crtp => "Crtp", // 126
+            &IpNextHeaderProtocols::Crudp => "Crudp", // 127
+            &IpNextHeaderProtocols::Sscopmce => "Sscopmce", // 128
+            &IpNextHeaderProtocols::Iplt => "Iplt", // 129
+            &IpNextHeaderProtocols::Sps => "Sps", // 130
+            &IpNextHeaderProtocols::Pipe => "Pipe", // 131
+            &IpNextHeaderProtocols::Sctp => "Sctp", // 132
+            &IpNextHeaderProtocols::Fc => "Fc", // 133
+            &IpNextHeaderProtocols::RsvpE2eIgnore => "RsvpE2eIgnore", // 134
+            &IpNextHeaderProtocols::MobilityHeader => "MobilityHeader", // 135
+            &IpNextHeaderProtocols::UdpLite => "UdpLite", // 136
+            &IpNextHeaderProtocols::MplsInIp => "MplsInIp", // 137
+            &IpNextHeaderProtocols::Manet => "Manet", // 138
+            &IpNextHeaderProtocols::Hip => "Hip", // 139
+            &IpNextHeaderProtocols::Shim6 => "Shim6", // 140
+            &IpNextHeaderProtocols::Wesp => "Wesp", // 141
+            &IpNextHeaderProtocols::Rohc => "Rohc", // 142
+            &IpNextHeaderProtocols::Test1 => "Test1", // 253
+            &IpNextHeaderProtocols::Test2 => "Test2", // 254*/
+            &IpNextHeaderProtocols::Reserved => "Reserved", // 255
+            _                          => "unknown"
```

Comment:
I originally generated all those constants from the list, so if you have them all (unless there are new entries), that should be all of them I think!